modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/config/io.scala

```diff
@@ -486,6 +486,24 @@ object io {
       deriveConfiguredEncoder[GcpUserAgent]
   }
 
+  case class BlobStorageClients(
+    gcs: Boolean,
+    s3: Boolean,
```

Comment:
Imagine we removed these booleans from the configuration.  What would be the disadvantage?

I guess you are trying to avoid creating clients if they will never be needed.  Especially as each client consumes JVM resources (threads, memory, etc).

If that is what you are trying to avoid....  Is it possible to only create the client **if** the `canDownload` matches.

For example, with `AzureStorageclient` you could move creating the store into the `download` method:

```scala
new Client[F] {
  def canDownload(uri: URI): Boolean = ???

  def download(uri: URI): Stream[F, Byte] =
    Stream.resource(createStore(storageName)).flatMap { store =>
      ???
    }
}
```

Reply:
Sounds like a good idea! I will start to make necessary changes.

Reply:
I'm OK with that.  Thanks for considering it.